docs/diagnostics/PX1063.md

```diff
@@ -0,0 +1,98 @@
+# PX1063
+This document describes the PX1063 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                                                         | Type                           | Code Fix    | 
+| ------ | ----------------------------------------------------------------------------------------- | ------------------------------ | ----------- | 
+| PX1063 | The declaration of the exception class does not contain a serialization constructor. This will cause an incorrect deserialization of the exception data and will lead to runtime errors in Acumatica ERP. | Error | Available | 
+
+## Diagnostic Description
+
+Every class that that is derived from the System.Exception class (including PXException-derived classes) must implement the serialization constructor.
+
+The code fix generates the serialization constructor for the exception class. The generated constructor will be private for sealed exception classes and protected for all other classes.  The generated constructor will call the base serialization constructor. 
+If the exception class declares a new serializable data then the constructor will contain a call to the `RestoreObjectProps` static method of the `PX.Common.ReflectionSerializer` class. This class is responsible for Acumatica custom reflection-based serialization and deserialization of data. If the constructor contains no serializable data, the serialization constructor body will be empty.
+
+By serializable data, the diagnostic understands any new data declared in the exception class that will be serialized by .Net serializers and Acumatica reflection based serializer which are the following:
+ - New fields if they are not marked with `System.NonSerializedAttribute` attribute:
+ - New auto generated properties if they are not marked with `System.NonSerializedAttribute` attribute:
+For example, the following types of members are not considered serializable:
```

Comment:
why not "members"?

Reply:
Because the listed cases in my opinion do not represent "types" of members. The member in C# is quire formal term, the possible types of a member are:
- property
- nested type - class, interface, enum, struct
- field,
- event
- method
- constructor,
- operator,
- deconstructor

May be there is something else but I think this covers most of the types. So the listed conditions are not types but just conditions on fields and properties. 
I also think, since the diagnostic works only on fields and properties, we should not generalize and use term `member` because this may cause confusion and people may think that, for example, methods are also somehow affected because methods are members too.